woocommerce/woocommerce-ios
# [REST API] Initial support for WPCom account creation during Jetpack setup

Closes: #14420
Closes: #14419


ℹ️ This is my first PR dealing with library updates, please let me know if I'm doing something wrong.

## Description
This PR adds initial support for creating accounts during the Jetpack setup, it's based on the changes of:
- WordPressKit: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/822
- WordPressAuthenticator: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/861

⚠️ Please don't merge until both PRs above are merged and the Podfile is updated.

With the above changes, we can now follow the same approach as the Jetpack web UI, if we detect that an account doesn't exist yet, we start the signup using a magic link, and when the user taps on it, they will then be automatically authenticated and they can continue the Jetpack setup.

The changes are behind a feature flag until we update the UI of the magic link screen with more clarifications about the signup flow.

## Steps to reproduce
1. Sign in to a site using Application passwords.
2. Start Jetpack setup from the dashboard or app settings.
3. Enter an email that's not registered yet on WordPress.com

## Testing information
- Confirm the above starts the magic link flow.
- Tap on the link in the email in your phone (or copy paste it from desktop to safari on your phone), and confirm the Jetpack setup is started after the authentication. 

## Screenshots
https://github.com/user-attachments/assets/82bc025d-50ca-4026-a7ca-a804a7867994

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
